File: src/test/java/me/saro/imw/jt/ImageConvertTest.java
package me.saro.imw.jt;

import me.saro.imw.ImageMagick;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.io.File;

import static org.junit.jupiter.api.Assertions.assertTrue;


@DisplayName("[Java] ImageMagick Image Convert Test")
public class ImageConvertTest {

    File root;
    File tmp;

    @Test
    @DisplayName("image info: example.gif")
    public void imageInfoGif() throws Exception {
        var info = ImageMagick.create().convert(new File(root,"example.gif"), new File(tmp, "example-gif.webp"));
        System.out.println(info);
        assertTrue(info.isSuccess());

        info = ImageMagick.create().resize(120, 120).webpCompression().convert(new File(root,"example.gif"), new File(tmp, "example-resize-gif.webp"));
        System.out.println(info);
        assertTrue(info.isSuccess());
    }

    @Test
    @DisplayName("image info: example.jpg")
    public void imageInfoJpg() throws Exception {
        var info = ImageMagick.create().convert(new File(root,"example.jpg"), new File(tmp, "example-jpg.webp"));
        System.out.println(info);
        assertTrue(info.isSuccess());

        info = ImageMagick.create().resize(120, 120).webpCompression().convert(new File(root,"example.jpg"), new File(tmp, "example-resize-jpg.webp"));
        System.out.println(info);
        assertTrue(info.isSuccess());
    }

    @Test
    @DisplayName("image info: example.png")
    public void imageInfoPng() throws Exception {
        var info = ImageMagick.create().convert(new File(root,"example.png"), new File(tmp, "example-png.webp"));
        System.out.println(info);
        assertTrue(info.isSuccess());

        info = ImageMagick.create().resize(120, 120).webpCompression().convert(new File(root,"example.png"), new File(tmp, "example-resize-png.webp"));
        System.out.println(info);
        assertTrue(info.isSuccess());
    }

    @BeforeEach
    public void bindPath() {
        try {
            root = new File(ImageConvertTest.class.getClassLoader().getResource("images").toURI());
            (tmp = new File("tmp")).mkdirs();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/test/kotlin/me/saro/imw/kt/ImageConvertTest.kt
package me.saro.imw.kt

import me.saro.imw.ImageMagick
import me.saro.imw.comm.ImageMagickResult
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import java.io.File

@DisplayName("[Koltin] ImageMagick Image Convert Test")
class ImageConvertTest {
    var root: File? = null
    var tmp: File? = null

    @Test
    @DisplayName("image info: example.gif")
    @Throws(Exception::class)
    fun imageInfoGif() {
        var info: ImageMagickResult<String> =
            ImageMagick.create().convert(File(root, "example.gif"), File(tmp, "example-gif.webp"))
        println(info)
        Assertions.assertTrue(info.isSuccess)

        info = ImageMagick.create().resize(120, 120).webpCompression()
            .convert(File(root, "example.gif"), File(tmp, "example-resize-gif.webp"))
        println(info)
        Assertions.assertTrue(info.isSuccess)
    }

    @Test
    @DisplayName("image info: example.jpg")
    @Throws(Exception::class)
    fun imageInfoJpg() {
        var info: ImageMagickResult<String> =
            ImageMagick.create().convert(File(root, "example.jpg"), File(tmp, "example-jpg.webp"))
        println(info)
        Assertions.assertTrue(info.isSuccess)

        info = ImageMagick.create().resize(120, 120).webpCompression()
            .convert(File(root, "example.jpg"), File(tmp, "example-resize-jpg.webp"))
        println(info)
        Assertions.assertTrue(info.isSuccess)
    }

    @Test
    @DisplayName("image info: example.png")
    @Throws(Exception::class)
    fun imageInfoPng() {
        var info: ImageMagickResult<String> =
            ImageMagick.create().convert(File(root, "example.png"), File(tmp, "example-png.webp"))
        println(info)
        Assertions.assertTrue(info.isSuccess)

        info = ImageMagick.create().resize(120, 120).webpCompression()
            .convert(File(root, "example.png"), File(tmp, "example-resize-png.webp"))
        println(info)
        Assertions.assertTrue(info.isSuccess)
    }

    @BeforeEach
    fun bindPath() {
        try {
            root = File(ImageConvertTest::class.java.classLoader.getResource("images").toURI())
            File("tmp").also { tmp = it }.mkdirs()
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }
}
